server/enhancedDate.js

```diff
@@ -1 +1,83 @@
-// export this as a Node module using the enhancedDate.js code from Module_2
\ No newline at end of file
+// export this as a Node module using the enhancedDate.js code from Module_2
+
+// code from module_2 modified for node use
+var enhancedDate = function() {
+    'use strict';
+
+    //create object
+    var myObject = {};
+    
+    //create variables
+    var daysOfWeek = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+    var monthNames = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
+  
+    myObject.setDate = function(passedDate) {
+        var now;
+
+        if (passedDate) {
+            if (passedDate instanceof Date) {
+                now = passedDate;
+            } else if (typeof (passedDate) === 'number') {
+                now = new Date(passedDate);
+            } else {
+                now = new Date();
+            }
+        }else {
+            //nothing passed in so get current datetime
+            now = new Date();
+        }
+
+        myObject.now = now;
+    };
+
+    myObject.getDate = function(wantObject) {
```

Comment:
All of the methods besides setDate should check that myObject.now isn't undefined before trying to use it. If it isn't, then create it at that point.